app/src/semaphore.h

```diff
@@ -50,29 +51,43 @@ class Semaphore {
 #if FIREBASE_PLATFORM_OSX || FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
     // MacOS requires named semaphores, and does not support unnamed.
     // Generate a unique string for the semaphore name:
-    static const char kPrefix[] = "/firebase-";
-    // String length of the name prefix.
-    static const int kPprefixLen = sizeof(kPrefix);
+
     // String length of the pointer, when printed to a string.
     static const int kPointerStringLength = 16;
     // Buffer size.  the +1 is for the null terminator.
-    static const int kBufferSize = kPprefixLen + kPointerStringLength + 1;
+    static const int kBufferSize = kPointerStringLength + 1;
 
     char buffer[kBufferSize];
-    snprintf(buffer, kBufferSize, "%s%016llx", kPrefix,
+    snprintf(buffer, kBufferSize, "%016llx",
              static_cast<unsigned long long>(  // NOLINT
                  reinterpret_cast<intptr_t>(this)));
 
-    semaphore_ = sem_open(buffer,
+#if FIREBASE_PLATFORM_OSX
+    // Sandbox mode is currently only available for macOS.
+    std::string semaphore_name = util::GetSandboxModeSemaphorePrefix();
+    if (semaphore_name.empty()) {
+      semaphore_name = "/firebase-";
```

Comment:
Is it possible to check at application start whether we are able to create a semaphore, and if not, print a loud fatal error to the log? (Similar to what happens with the iOS SDK when you don't have a REVERSED_CLIENT_ID URL handler set up.)

Reply:
Do you have a link to that check?

Reply:
https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/AuthProvider/OAuth/FIROAuthProvider.m#L139

Reply:
Got it, thanks. I thought the example would have been more tightly integrated into the Firebase App initialization so I could use it as an example.

Trying my own thing. We'll see how it works out.